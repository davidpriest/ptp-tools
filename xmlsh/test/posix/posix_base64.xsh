# Test base command
import package posix=org.xmlsh.commands.posix

rm -rf $TMPDIR/_xmlsh
mkdir $TMPDIR/_xmlsh


# Test 1 simple base64 encode and decode of "Hello World"

echo "Hello Word" | posix:base64 | posix:base64 -d 

# Use a known binary file for testing
posix:cp ../../lib/saxon9he.jar $TMPDIR/_xmlsh/binary.dat
cd $TMPDIR/_xmlsh
xmd5sum binary.dat
posix:base64 -w binary.dat > binary.b64
posix:base64 -d binary.b64 > b2.dat
xmd5sum b2.dat

cd ..
rm -rf $TMPDIR/_xmlsh
