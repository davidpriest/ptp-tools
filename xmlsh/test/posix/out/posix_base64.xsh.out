Hello Word
<xmd5>
   <file name="binary.dat"
         md5="f6a7134b8debde6a76a4f037d92d4754"
         length="9788989"/>
</xmd5>
<xmd5>
   <file name="b2.dat"
         md5="f6a7134b8debde6a76a4f037d92d4754"
         length="9788989"/>
</xmd5>
